goon/code/datums/browserOutput.dm

```diff
@@ -131,6 +133,9 @@ GLOBAL_DATUM_INIT(iconCache, /savefile, new("tmp/iconCache.sav")) //Cache of ico
 	if(volume)
 		adminMusicVolume = CLAMP(text2num(volume), 0, 100)
 
+/datum/chatOutput/proc/sendEmojiList()
+	ehjax_send(data = GLOB.emojiJson)
```

Comment:
Yeah, this should be using the asset cache

Reply:
> Yeah, this should be using the asset cache

Why? It is needed only for the server to loaded list of emoji. The player does not need this list.

Reply:
It's for caching and efficiency. Base64 is very laggy for images and it was the main reason making it unplayable with lots of messages around.

Reply:
> It's for caching and efficiency. Base64 is very laggy for images and it was the main reason making it unplayable with lots of messages around.

I checked on RU-TGMC with online 40+ and emoji spam, there were no lags.

Reply:
Yes, but what if you get 100 players, and people talking in OOC using emojis often?
The solution for the problem already exists, and just because right now it doesn't have much impact doesn't mean it never will. Switching it to asset cache shouldn't be that hard, you'll just have to make it a spritesheet, then you'll have to look up how the icon2html proc embeds icons and use that to display the emojis.

Reply:
> Yes, but what if you get 100 players, and people talking in OOC using emojis often?
> The solution for the problem already exists, and just because right now it doesn't have much impact doesn't mean it never will. Switching it to asset cache shouldn't be that hard, you'll just have to make it a spritesheet, then you'll have to look up how the icon2html proc embeds icons and use that to display the emojis.

Good asset cashe would be a better option. Why redo processing smiles if they are already processed in GoonChat?

Reply:
Because goonchat is using base64 each time to display them, and that has the aforementioned issues.

Reply:
> Because goonchat is using base64 each time to display them, and that has the aforementioned issues.

So if I add pictures to asset cashe, then I can simply make png instead of references to Base64.